jshell can't see dependencies

with 

```
///usr/bin/env jbang "$0" "$@" ; exit $?
//DEPS https://github.com/blocoio/faker/tree/1.2.8

import io.bloco.faker.Faker;

Faker fake = new Faker("da-DK");
IntStream.range(0, 10).forEach(
           i-> System.out.println(fake.name.name())
        );
```

I get

```java
 jbang --verbose test.jsh
[jbang] jbang version 0.79.0
[jbang] System Java version detected as 11
[jbang] System Java version matches requested version 11
[jbang] System Java version matches requested version 11
[jbang] run: /Users/max/.sdkman/candidates/java/current/bin/jshell --class-path=/Users/max/.m2/repository/com/github/blocoio/faker/1.2.8/faker-1.2.8.jar:/Users/max/.m2/repository/com/github/bmoliveira/snake-yaml/v1.18-android/snake-yaml-v1.18-android.jar:/Users/max/.m2/repository/joda-time/joda-time/2.9.1/joda-time-2.9.1.jar --startup=DEFAULT --startup=/var/folders/yb/sytszfld4sg8vwr1h0w20jlw0000gn/T/jbang_arguments_15400451799524567493test.jsh /Users/max/code/personal/test.jsh /var/folders/yb/sytszfld4sg8vwr1h0w20jlw0000gn/T/jbang_exit_8285782591836528220test.jsh
Error:
cannot find symbol
  symbol:   class Fake
import io.bloco.faker.Fake;
       ^-----------------^
Error:
cannot find symbol
  symbol:   class Faker
Faker fake = new Faker("da-DK");
^---^
Error:
cannot find symbol
  symbol:   class Faker
Faker fake = new Faker("da-DK");
                 ^---^
Error:
package fake does not exist
           i-> System.out.println(fake.name.name())
                                  ^-------^
```

which is weird since this works:

```java
///usr/bin/env jbang "$0" "$@" ; exit $?
// //DEPS <dependency1> <dependency2>
//DEPS https://github.com/blocoio/faker/tree/1.2.8

import static java.lang.System.*;

import java.util.stream.IntStream;

import io.bloco.faker.*;

public class test {

    public static void main(String... args) {
        Faker fake = new Faker("da-DK");
        IntStream.range(0, 10).forEach(
           i-> out.println(fake.name.name())
        );
    }
}
```

So it's the latest `v0.88.0` (so it might not be the same issue as the original one...):
```
$ ./jbang-0.87.0/bin/jbang https://raw.githubusercontent.com/tadayosi/jbang-sandbox/main/camel/camel.jsh
Running Camel route...
Hello Camel!
Hello Camel!
Hello Camel!
```
```
$ ./jbang-0.88.0/bin/jbang https://raw.githubusercontent.com/tadayosi/jbang-sandbox/main/camel/camel.jsh
Running Camel route...
Exception java.lang.NoClassDefFoundError: org/apache/camel/main/Main
      at (#8:1)
Caused by: java.lang.ClassNotFoundException: org.apache.camel.main.Main
      at URLClassLoader.findClass (URLClassLoader.java:476)
      at DefaultLoaderDelegate$RemoteClassLoader.findClass (DefaultLoaderDelegate.java:154)
      at ClassLoader.loadClass (ClassLoader.java:589)
      at ClassLoader.loadClass (ClassLoader.java:522)
      ...
Exception java.lang.NoClassDefFoundError: org/apache/camel/main/Main
      at (#10:1)
Caused by: java.lang.ClassNotFoundException: org.apache.camel.main.Main
      at URLClassLoader.findClass (URLClassLoader.java:476)
      at DefaultLoaderDelegate$RemoteClassLoader.findClass (DefaultLoaderDelegate.java:154)
      at ClassLoader.loadClass (ClassLoader.java:589)
      at ClassLoader.loadClass (ClassLoader.java:522)
```

But again, even the fixed `test.jsh` doesn't run with `v0.88.0`:
```
$ cat test.jsh 
///usr/bin/env jbang "$0" "$@" ; exit $?
//DEPS https://github.com/blocoio/faker/tree/1.2.8

import io.bloco.faker.Faker;

Faker fake = new Faker("da-DK");
IntStream.range(0, 10).forEach(
           i-> System.out.println(fake.name.name())
        );
```
```
$ ./jbang-0.88.0/bin/jbang test.jsh 
Exception java.lang.NoClassDefFoundError: io/bloco/faker/Faker
      at (#2:1)
Caused by: java.lang.ClassNotFoundException: io.bloco.faker.Faker
      at URLClassLoader.findClass (URLClassLoader.java:476)
      at DefaultLoaderDelegate$RemoteClassLoader.findClass (DefaultLoaderDelegate.java:154)
      at ClassLoader.loadClass (ClassLoader.java:589)
      at ClassLoader.loadClass (ClassLoader.java:522)
      ...
Exception java.lang.NoClassDefFoundError: io/bloco/faker/Faker
      at lambda$do_it$$0 (#3:2)
      at Streams$RangeIntSpliterator.forEachRemaining (Streams.java:104)
      at IntPipeline$Head.forEach (IntPipeline.java:593)
      at (#3:1)
Caused by: java.lang.ClassNotFoundException: io.bloco.faker.Faker
      at URLClassLoader.findClass (URLClassLoader.java:476)
      at DefaultLoaderDelegate$RemoteClassLoader.findClass (DefaultLoaderDelegate.java:154)
      at ClassLoader.loadClass (ClassLoader.java:589)
      at ClassLoader.loadClass (ClassLoader.java:522)
      ...
```
it works with `v0.87.0`.

--- a/itests/deps.jsh
+++ b/itests/deps.jsh
@@ -1,0 +1,7 @@
+///usr/bin/env jbang "$0" "$@" ; exit $?
+//DEPS https://github.com/blocoio/faker/tree/1.2.8
+
+import io.bloco.faker.Faker;
+
+Faker fake = new Faker("da-DK");
+System.out.println(fake.name.name());

--- a/itests/jsh.feature
+++ b/itests/jsh.feature
@@ -37,3 +37,12 @@ Scenario: jsh sources
   When command('jbang main.jsh')
   Then match err == ""
   Then match out == "hello\n"
+
+Scenario: jsh with deps 1
+  When command('jbang deps.jsh')
+  Then match err == ""
+  Then match out == "Hello World\n"
+
+Scenario: jsh with deps 2
+  When command('jbang deps.jsh')
+  Then match err == ""

--- a/src/main/java/dev/jbang/cli/Run.java
+++ b/src/main/java/dev/jbang/cli/Run.java
@@ -194,10 +194,15 @@ public class Run extends BaseBuildCommand {
 					}
 				}
 
+				// NB: See https://github.com/jbangdev/jbang/issues/992 for the reasons why we
+				// use the -J flags below
+
 				optionalArgs.add("--execution=local");
+				optionalArgs.add("-J--add-modules=ALL-SYSTEM");
 
 				if (!Util.isBlankString(classpath)) {
 					optionalArgs.add("--class-path=" + classpath);
+					optionalArgs.add("-J--class-path=" + classpath);
 				}
 
 				optionalArgs.add("--startup=DEFAULT");

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -161,7 +161,8 @@ public class TestRun extends BaseTest {
 
 		String result = run.generateCommandLine(src, ctx);
 
-		assertThat(result, matchesPattern("^.*jshell(.exe)? --execution=local --startup.*$"));
+		assertThat(result,
+				matchesPattern("^.*jshell(.exe)? --execution=local -J--add-modules=ALL-SYSTEM --startup.*$"));
 		assertThat(result, not(containsString("  ")));
 		assertThat(result, containsString("helloworld.jsh"));
 		assertThat(result, not(containsString("--source 11")));
@@ -227,7 +228,8 @@ public class TestRun extends BaseTest {
 
 		String result = run.generateCommandLine(src, ctx);
 
-		assertThat(result, matchesPattern("^.*jshell(.exe)? --execution=local --startup.*$"));
+		assertThat(result,
+				matchesPattern("^.*jshell(.exe)? --execution=local -J--add-modules=ALL-SYSTEM --startup.*$"));
 		assertThat(result, not(containsString("  ")));
 		assertThat(result, containsString("empty.jsh"));
 		assertThat(result, not(containsString("--source 11")));
@@ -249,7 +251,8 @@ public class TestRun extends BaseTest {
 
 		String result = run.generateCommandLine(src, ctx);
 
-		assertThat(result, matchesPattern("^.*jshell(.exe)? --execution=local --startup.*$"));
+		assertThat(result,
+				matchesPattern("^.*jshell(.exe)? --execution=local -J--add-modules=ALL-SYSTEM --startup.*$"));
 		assertThat(result, not(containsString("  ")));
 		assertThat(result, containsString("hellojsh"));
 		assertThat(result, not(containsString("--source 11")));
@@ -455,7 +458,8 @@ public class TestRun extends BaseTest {
 
 		String result = run.generateCommandLine(src, ctx);
 
-		assertThat(result, matchesPattern("^.*jshell(.exe)? --execution=local --startup.*$"));
+		assertThat(result,
+				matchesPattern("^.*jshell(.exe)? --execution=local -J--add-modules=ALL-SYSTEM --startup.*$"));
 		assertThat(result, containsString("funcs.jsh"));
 		assertThat(result, containsString("main.jsh"));
 		assertThat(result, containsString("--startup=DEFAULT"));

